### PR TITLE
Update ACI liveness/readiness probe handling to work with named ports

### LINKED ISSUE #331: Named port resolution doesn't work
---

### Environment summary

Provider (e.g. ACI, AWS Fargate, Hyper)

ACI

Version (e.g. 0.1, 0.2-beta)

K8s Master Info (e.g. AKS, ACS, Bare Metal, EKS)

AKS

Install Method (e.g. Helm Chart, )

Helm Chart
### Issue Details

A Pod spec with a named port yields a 500 on ACI.

```
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
          livenessProbe:
            httpGet:
              path: /
              port: http
          readinessProbe:
            httpGet:
              path: /
              port: http
```

Changing the ports to the container port number resolves the issue. In the API request sent to ACI, the port number is set to 0 until the liveness and readiness probes.

### Repo Steps

Install VK w ACI using Helm
Create a deployment with named port references in liveness and readiness checks scheduled on a ACI backed VK

### PR DESCRIPTION
**Pull request by jeremyrickard**
This updates the ACI provider to check if a probe uses named ports or numeric values. Previously, it only attempted to use numeric values.

Fixes #331

### CODE REVIEW

**Comment by cpuguy83** [comment]
Hmm, stats test failed for some reason. Doesn't seem like this should have any affect on it.

**Comment by jeremyrickard** [comment]
@cpuguy83 merged latest into this and that retriggered a clean run of the tests. Should be ready to be reviewed.

**Comment by cpuguy83** [comment]
```
providers/azure/aci_test.go:585:31: not enough arguments in call to provider.CreatePod
	have (*"github.com/virtual-kubelet/virtual-kubelet/vendor/k8s.io/api/core/v1".Pod)
	want (context.Context, *"github.com/virtual-kubelet/virtual-kubelet/vendor/k8s.io/api/core/v1".Pod)
```

**Comment by srrengar** [comment]
@jeremyrickard can you look into the test failures to close the PR and we can close #331 ?

File: providers/azure/aci.go [hunk 1]
```diff
@@ -32,8 +32,9 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
 	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
-	"k8s.io/client-go/tools/remotecommand"
+  "k8s.io/client-go/tools/remotecommand"
```
Comment (pires) [inline providers/azure/aci.go]: `go fmt` will complain here.

**Comment by srrengar** [comment]
@VDuleb can you look into this

**Comment by jeremyrickard** [comment]
I'll pick this back up and resolve the conflicts and address the above.

File: providers/azure/aci.go [hunk 2]
```diff
@@ -1298,8 +1298,26 @@ func getProbe(probe *v1.Probe) (*aci.ContainerProbe, error) {
...
+		port := probe.Handler.HTTPGet.Port
+		switch port.Type {
+		case intstr.Int:
+			portValue = port.IntValue()
+		case intstr.String:
+			portValue = -1
+			portName := port.String()
+			for _, p := range ports {
+				if portName == p.Name {
+					portValue = int(p.ContainerPort)
```
Comment (cpuguy83) [inline providers/azure/aci.go]: break?

File: providers/azure/aci.go [hunk 3]
```diff
@@ -1298,8 +1298,26 @@ func getProbe(probe *v1.Probe) (*aci.ContainerProbe, error) {
 
 	var httpGET *aci.ContainerHTTPGetProbe
 	if probe.Handler.HTTPGet != nil {
+		var portValue int
+		port := probe.Handler.HTTPGet.Port
+		switch port.Type {
+		case intstr.Int:
+			portValue = port.IntValue()
+		case intstr.String:
+			portValue = -1
```
Comment (cpuguy83) [inline providers/azure/aci.go]: Is 0 a valid port value? Can we use the 0 value if not?

Comment (jeremyrickard) [reply providers/azure/aci.go]: 👍

File: providers/azure/aci.go (hunk 2 above)
Comment (jeremyrickard) [reply providers/azure/aci.go]: Good catch, 👍

**Comment by jeremyrickard** [comment]
@cpuguy83 fixed up your comments, take another look please :)

**Pull request merged**